add newline in error message between code path and actual failed line in test file

Currently, data driven tests return an error message that looks like:
```
    --- FAIL: TestBackupRestoreDataDriven/column-families (7.88s)
        backup_test.go:283: 
            /private/var/tmp/_bazel_michaelbutler/13ba282fa2b19539d0c969c1113bb37c/sandbox/darwin-sandbox/1/execroot/cockroach/bazel-out/darwin-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/cockroach/pkg/ccl/backupccl/testdata/backup-restore/column-families:52: SELECT max(length(start_key)) FROM [SHOW RANGES FROM TABLE r1.cfs];
            expected:
            2
            
            found:
            <nil>
    backup_test.go:379: -- test log scope end --
FAIL
I220324 21:54:59.442315 1 (gostd) testmain.go:432  [-] 1  Test //pkg/ccl/backupccl:backupccl_test exited with error code 1


ERROR: <nil>

202 runs completed, 1 failures, over 2m37s
context canceled
FAIL
```
A more user friendly error would add a newline between the code path and the actual line that failed, i.e. (you can see the SELECT statement in question below): 
```
    --- FAIL: TestBackupRestoreDataDriven/column-families (7.88s)
        backup_test.go:283: 
            /private/var/tmp/_bazel_michaelbutler/13ba282fa2b19539d0c969c1113bb37c/sandbox/darwin-sandbox/1/execroot/cockroach/bazel-out/darwin-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/cockroach/pkg/ccl/backupccl/testdata/backup-restore/column-families:52: 
           SELECT max(length(start_key)) FROM [SHOW RANGES FROM TABLE r1.cfs];
            expected:
            2
            
            found:
            <nil>
    backup_test.go:379: -- test log scope end --
FAIL
I220324 21:54:59.442315 1 (gostd) testmain.go:432  [-] 1  Test //pkg/ccl/backupccl:backupccl_test exited with error code 1


ERROR: <nil>

202 runs completed, 1 failures, over 2m37s
context canceled
FAIL
```

The lack of newline  lead to some confusion diagnosing a data driven test failure: [here](https://github.com/cockroachdb/cockroach/issues/77264#issuecomment-1065010543) and [here](https://github.com/cockroachdb/cockroach/issues/77264#issuecomment-1077884882).

--- a/datadriven.go
+++ b/datadriven.go
@@ -361,12 +361,12 @@ func runDirective(t *testing.T, r *testDataReader, f func(*testing.T, *TestData)
 				B:       actualLines,
 			})
 			if err == nil {
-				t.Fatalf("\n%s: %s\noutput didn't match expected:\n%s", d.Pos, d.Input, diff)
+				t.Fatalf("\n%s:\n %s\noutput didn't match expected:\n%s", d.Pos, d.Input, diff)
 				return
 			}
 			t.Logf("Failed to produce diff %v", err)
 		}
-		t.Fatalf("\n%s: %s\nexpected:\n%s\nfound:\n%s", d.Pos, d.Input, d.Expected, actual)
+		t.Fatalf("\n%s:\n %s\nexpected:\n%s\nfound:\n%s", d.Pos, d.Input, d.Expected, actual)
 	} else if Verbose() {
 		input := d.Input
 		if input == "" {

